plugin/pkg/shoot/validator/admission.go

```diff
@@ -365,7 +365,29 @@ func (c *validationContext) validateScheduling(a admission.Attributes, shootList
 		}
 
 		if oldSeed.Spec.Provider.Type != c.seed.Spec.Provider.Type {
-			return admission.NewForbidden(a, fmt.Errorf("cannot change Seed because cloud provider for new seed (%s) is not equal to cloud provider for old seed (%s)", c.seed.Spec.Provider.Type, oldSeed.Spec.Provider.Type))
+			return admission.NewForbidden(a, fmt.Errorf("cannot change seed because cloud provider for new seed (%s) is not equal to cloud provider for old seed (%s)", c.seed.Spec.Provider.Type, oldSeed.Spec.Provider.Type))
+		}
+
+		if c.shoot.Status.LastOperation != nil && c.shoot.Status.LastOperation.State != core.LastOperationStateSucceeded {
+			return admission.NewForbidden(a, fmt.Errorf("cannot change seed name because the last operation has not finished successfully yet: operation %s, state %s", c.shoot.Status.LastOperation.Type, c.shoot.Status.LastOperation.State))
+		}
```

Comment:
This is a quite harsh limitation I would say, especially for the bad case I would guess.

> [...] which means that they will be applied during restoration.

Why would this be problematic?

Reply:
I wondered a lot about this one. My worries currently are due to the following scenario: 
1. Someone changes the kubernetes version and reconciliation starts
2. Then the seedName is changed
3. Then for some reason before the new kubernetes version is applied, gardenlet is restarted (or there is an error during reconciliation and reconciliation is restarted)
4. This will cause migration to start which (when I tested it locally) leaves the nodes with the old kubernetes version orphaned.

I also considered that we could introduce a `force` annotation to force control plane migration even if the last operation is not successful yet.

Reply:
I don't know, honestly, this sounds highly artificial given that migration is no every day-to-day operation. And even if the nodes remain orphaned, so what? MCM will clean them up eventually since there are no backing `Machine` objects anymore, wouldn't it?

Reply:
You're right that it probably would happen very rarely, and that if MCM works properly it should clean up the orphaned nodes. But I am afraid what happens if the cluster has a lot of nodes, as i'm not really sure if they will be rolled properly or all orphaned nodes will get immediately deleted and the workload on them moved to the new nodes. This could cause an unwanted downtime.

Reply:
Well, I guess even with this validation there will be corner cases for such things to happen. Anyways, you want to keep it, well aware that this might impact usability of the bad case (I assume, when the source seed is done, the `lastOperation` might have a higher probability of not being `Succeeded`)?

Reply:
I'm only willing to keep it if we introduce a some type of "confirmation" or "force" annotation, that would allow such changes when the last operation is not succeeded.

Reply:
I have mixed feelings (rather negative) regarding introducing such additional code/complexity for a corner case that is highly unlikely to hit.

Reply:
Since I was also generally doubting it, let's skip the check, and only think about adding it if we ever start to frequently hit such corner cases. I've removed it in my latest commit.